### ISSUE TITLE
Angular2 known issues

### ISSUE THREAD

**Issue by deepu105** [jhipster/generator-jhipster#4407]
Adding a known issues todo list for Angular2 so that we have track of these here
- [x] `Resetting invalid cookie language "en" to prefered language "en"` this is being logged enormous times, thats not the issue, the issue is somewhere the stuff is being run so many times then really needed and its kind of slowing down the page loads. This needs to be investigated and fixed
- [x] There is a console error during first login after a page refresh
- [x] UI bootstrap doesnt seem to work -migrated to ng-bootstrap
- [x] UI router doesnt seem to work completeley - moved to ui-router-ng2 see https://github.com/jhipster/generator-jhipster/pull/4220
- [x] NPM `postinstall` of `typings install` doesn't work in windows (not sure of other os) some weird error. It works when the command is run manually - **works** `typings` needs to be installed with global scope `npm i -g typings` [see](https://github.com/jhipster/generator-jhipster/issues/3664#issuecomment-243690915)
- [x] Health modal needs to be migrated
- [x] Update to RC6
- [x] After RC6 update: This app works for now. But we have an issue. If we use multiple `@NgModule` like how I have defined for now, then we are not able to use the common pipes and directives across these, seems like app doesnt load if I declare a pipe in more than one module. we need to find a solution for this or we need to stick to just a single huge module for the entire app(Which is ugly) - **fixed** [see](https://github.com/jhipster/generator-jhipster/issues/3664#issuecomment-245885129)
- [x] click of any header menu item reloads the page [after ng-bootstrap update] **fixed** added `href="#"` to empty anchors
- [x] shared directives needs to be tested and validated
- [x] error when clicking on the login link in account/register page
- [x] swagger api page doesnt load
- [x] websocket connection is failing
- [x] User Management Detail/Delete Dialogs don't load translation into page title
- [x] User Management user list does not refresh after making a change to the users
- [x] User Management pagination doesnt work properly
- [x] metrics page needs update once #4454 is merged to master
- [x] password strength bar directive doesnt work
- [x] The navbar menu doesnt work in mobile view
- [x] bootstrap show/hide elements based on screensize doesnt work might be a class rename issue
- [x] missing translations in health screen
- [X] user management view details is very slow to load
- [x] websocket sendActivity is not being triggered, need to see if can be triggered on route changes
- [x] tracker page needs to be tested for edge cases
- [x] There is an error from ui-router sometimes during session timeout `Transition #0 Digest #47: <- Rejected "Transition#0( ''{} -> 'password'{} )", reason: TransitionRejection(type: 2, message: The transition has been superseded by a different transition, detail: Transition#1( ''{} -> 'accessdenied'{} ))
  angular-ui-router.js:5861 TransitionRejection(type: 2, message: The transition has been superseded by a different transition, detail: Transition#1( ''{} -> 'accessdenied'{} ))$defaultErrorHandler @ angular-ui-router.js:5861(anonymous function) @ angular-ui-router.js:6184processQueue @ angular.js:15961(anonymous function) @ angular.js:15977$eval @ angular.js:17229$digest @ angular.js:17045(anonymous function) @ angular.js:17268completeOutstandingRequest @ angular.js:5869(anonymous function) @ angular.js:6145ZoneDelegate.invokeTask @ zone.js:225Zone.runTask @ zone.js:125ZoneTask.invoke @ zone.js:293`
- [x] JHI Item Sort doesn't function
- [x] test sass handling when option is enabled
- [x] add uiSrefActive back once the related issue is fixed upstream
- [x] favicon doesnt load

**Comment by deepu105** [comment]
@ruddell can check the translation issue again? for me its loading translations

**Comment by ruddell** [comment]
@deepu105  The translations on the page work but the titles aren't loaded: `translation-not-found[user-management.detail.title]`.  This also affects the main user-management page.

**Comment by clp334264003** [comment]
I can not get the current user from the server.and The message header has no token
get(): Observable<any> {
    return this.http.get('api/account').map((res: Response) => res.json());
  }

**Comment by mraible** [comment]
Translation seems to work for me. Login does not. I click on "Sign in" and nothing happens. Also, the following import statement does not resolve classes in IntelliJ IDEA:

```
import { Account, LoginModalService, Principal } from "../shared";
```

This does work:

```
import { Account, LoginModalService, Principal } from "../shared/index";
```

Should I change these?

**Comment by deepu105** [comment]
Matt its coz of IDE issue. It works fine in vscode and while doing tscompile

**Comment by deepu105** [comment]
The sign in on menu will work, but the one in the page doesn't work coz
translated content is sanitized and those directive are stripped. I still
don't have a solution for that

**Comment by aldranvijay** [comment]
I am following instruction available on this document : https://github.com/jhipster/generator-jhipster/blob/master/NG2-GUIDE.md
But always getting the following error : 
![error](https://cloud.githubusercontent.com/assets/9549492/21447186/1398d468-c8f7-11e6-99a9-477c266aa42c.PNG)

Any help is appreciated :)

**Comment by pascalgrimaud** [comment]
Not sure this guide is up-to-date because the angular-2 branch has been merged into master and is deleted.

Simply use the branch master.
If you still have issue, plz provide as much as information, so we can help you

**Comment by aldranvijay** [comment]
Yes I am using the master branch source code. With the following given steps  in order : 
1. npm link 
2. yo jhipster (with npm install)
3. ./mvnw
4. npm run webpack:dev

**Comment by pascalgrimaud** [comment]
@albertmatyi : here all the steps

Clone the generator-jhipster and use npm link:
- `git clone https://github.com/jhipster/generator-jhipster.git`
- `cd generator-jhipster`
- `npm install`
- `npm link`

Create the project:
- `mkdir tmp && cd tmp`
- `npm link generator-jhipster`
- `yo jhipster`

**Comment by gmarziou** [comment]
And you may replace `npm` by `yarn` in the instructions above for better performance and reliability.

**Comment by sendilkumarn** [comment]
[this](https://goo.gl/cix8BY) argues that the elements should have a `custom` tag before. 

How about we adding `appname` as prefix to all the components or `jhi` ?

**Comment by deepu105** [comment]
@jhipster/developers I confirm the issue by @aldranvijay above. I'm gonna investigate this further

**Comment by pascalgrimaud** [comment]
@deepu105 : maybe you have the same screen because of spring boot 1.4.3. I revert to 1.4.2, and didn't manage to reproduce the issue reported from @aldranvijay

**Comment by deepu105** [comment]
i can reproduce issue on master with both ng1 and ng2 so definetely something from backend. ill try reverting boot

**Comment by deepu105** [comment]
@sendilkumarn may be we could use the jhiPrefix option for this

**Comment by aldranvijay** [comment]
Hi @deepu105 ,

After the application generation getting the following log : 

```| +-- window-size@0.2.0
| +-- y18n@3.2.1
| `-- yargs-parser@4.2.0
|   `-- camelcase@3.0.0
`-- UNMET PEER DEPENDENCY zone.js@0.7.2

npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.15
npm WARN @angular/http@2.2.3 requires a peer of rxjs@5.0.0-beta.12 but none wasinstalled.
npm WARN @angular/core@2.2.3 requires a peer of rxjs@5.0.0-beta.12 but none wasinstalled.
npm WARN @angular/core@2.2.3 requires a peer of zone.js@^0.6.21 but none was installed.
npm WARN string-replace-webpack-plugin@0.0.4 requires a peer of webpack@^1.4.2 but none was installed.
Server app generated successfully.
Client app generated successfully.
````

**Comment by deepu105** [comment]
thats normal.


Thanks & Regards,
Deepu

**Comment by sendilkumarn** [comment]
@deepu105 I will do a PR this week.

**Comment by VictorADS** [comment]
Can I have more information on this one ? 

> There is an error from ui-router sometimes during session timeout

**Comment by jdubois** [comment]
Just checked the box for "User Management pagination doesnt work properly" -> I tried with 50 users, no problem for me

**Comment by jdubois** [comment]
Checked the box for "add uiSrefActive back once the related issue is fixed upstream" as we do have the attribute, and it's working fine in the navbar

**Comment by deepu105** [comment]
I think the uiSrefActive thing was for the main navbar item highlight when
a submenu is selected. It doesnt seem to work now. There was a ui-router
bug for that


Thanks & Regards,
Deepu

**Comment by jdubois** [comment]
@deepu105 you mean, "it does seem to work now", right?

**Comment by deepu105** [comment]
No it doesnt work.

Say for example go to the metrics screen and look at the admin menu its not
highlighted. selected menu and parent needs to be highlighted. ui router
has a contains directive or somthing for that


Thanks & Regards,
Deepu

**Comment by jdubois** [comment]
Oh sorry @deepu105 I was just looking at the sub menu, not the main menu... Unchecking....

**Comment by antid0te** [comment]
Hi, i have a blank page when running `npm run webpack:dev`.
I watch at the webpack's config, from my understanding, you are using browser-sync-webpack-plugin but it is either not configured to forward request to webpack devServer or to serve resource directly. 
Is it a miss, a bug or i totally dont understand your config ?

**Comment by VictorADS** [comment]
@antid0te You need to use` ./mvnw` before running `npm run webpack:dev` and then you can access your site at localhost:9000

**Comment by antid0te** [comment]
@antid0te Thanks, it's working now. There is a bug with translation if you dont check 'enable internationalization support' in the generation wizard.

`Unhandled Promise rejection: Template parse errors:
Can't bind to 'jhi-translate' since it isn't a known property of 'span'. ("="text-xs-center">
                        <span class="tag" [ngClass]="getTagClass(health.status)" [ERROR ->]jhi-translate="{{'health.status.' + health.status}}">
                            {{health.status}}
 "): JhiHealthCheckComponent@21:81 ; Zone: <root> ; Task: Promise.then ; Value: SyntaxError Error: Template parse errors:
Can't bind to 'jhi-translate' since it isn't a known property of 'span'. ("="text-xs-center">
                        <span class="tag" [ngClass]="getTagClass(health.status)" [ERROR ->]jhi-translate="{{'health.status.' + health.status}}">
                            {{health.status}}
 "): JhiHealthCheckComponent@21:81`

I regenerated the project with internationalization and it's working with it.

**Comment by VictorADS** [comment]
Can you link me your .yo-rc.json pls  ?

**Comment by antid0te** [comment]
Weird, i try to reproduce the problem by regenerate the project with the options i picked, but i can't reproduce it.

**Comment by VictorADS** [comment]
Okay I think I know what's the problem. When we have no translation we should use j`hi-translate="'health.status.' + health.status"` and with translation it should be `jhi-translate="{{'health.status.' + health.status}}"`

**Comment by deepu105** [comment]
@sendilkumarn @erkieh I tried both your version of jhiSort, jhiSortBy directives but both doesnt seem to work. I tried in user management screen. can you guys check? check the latest version in ng-jhipster lib

**Comment by sendilkumarn** [comment]
@deepu105 remove the callback function and check. The UI works. But there is somwthing that has to be changes in ui-router's trsnsition.

**Comment by aldranvijay** [comment]
Getting following error on **yarn install** : 
```` 
error angular2-cookie@1.2.6: The engine "node" is incompatible with this module.
 Expected version ">=5.0.0".
error Found incompatible module
````
Is it normal ?

**Comment by deepu105** [comment]
You seem to have old node version try with 6+ LTS

Thanks & regards,
Deepu

**Comment by jdubois** [comment]
I checked the "favicon" item because of https://github.com/jhipster/generator-jhipster/commit/36216381cdcfcc90238804b6263cd64d1b9a751a -> not a very pretty way to correct this, but it works

**Comment by jdubois** [comment]
I checked the "sass" item: just tested it, it works very well

**Comment by jdubois** [comment]
Checked the "sort" item because of #4893

**Comment by gzsombor** [comment]
Checked the 3 tracker issue, it's seems to be working - the logout problem was fixed by #4908, the other two were #4819 I believe.

**Comment by jdubois** [comment]
Thanks @gzsombor !! As everything is checked, I'm closing this :-)

**Comment by cloud4288** [comment]
Any updates on the release candidate?
